teslajsonpy/controller.py

```diff
@@ -732,8 +736,9 @@ def get_homeassistant_components(self):
 
     def _add_energysite_components(self, energysite):
         self.__components.append(SolarPowerSensor(energysite, self))
-        self.__components.append(LoadPowerSensor(energysite, self))
-        self.__components.append(GridPowerSensor(energysite, self))
+        if energysite["components"]["load_meter"]:
```

Comment:
I thought about using the get method but I kind of like having a KeyError assert here and in certain parts of the code because I think it will quickly highlight Tesla has changed their API (which seems like is changing a bit here and there on the energy side). I can definitely use the get method though if you think that’s more appropriate here.

Reply:
I think users prefer graceful failure on api changes. A key error can potentially break unrelated parts of the API depending on where the error is raised. This particular error wasn't an api change but a system configuration you didn't have access to.

If you really want to know about these situations, set an else when the check fails logging a warning and ask for a issue report.  The warning will appear in user logs and I've found there's enough paranoid users who want clean logs that you'll see it.

Reply:
Sounds good, just pushed a change with the `get` method.